src/Npgsql/NpgsqlConnector.cs

```diff
@@ -957,6 +959,13 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
                     }
                     throw;
                 }
+                catch (NpgsqlException)
+                {
+                    // An ErrorResponse isn't followed by ReadyForQuery
+                    if (error != null)
+                        ExceptionDispatchInfo.Capture(error).Throw();
```

Comment:
Why not apply this also to the regular re-throwing of the error when ReadyForQuery is received? There's a `throw error` there as well.

Reply:
Thank you for reviewing.

When ReadyForQuery is received, no exception has occurred. So I think the regular-rethrowing does not need to apply `ExceptionDispatchInfo`. Is it right?

However, I am not familiar with `ExceptionDispatchInfo` so much.

Reply:
Never mind, it's not really important - it could possibly make sure we get a stack trace from the BackendMessageCode.ErrorResponse switch case instead of from the BackendMessageCode.ReadyForQuery switch case, but it's insignificant.